python/llm/src/bigdl/llm/transformers/models/utils.py

```diff
@@ -68,6 +68,14 @@ def apply_rotary_pos_emb(q, k, cos, sin, position_ids, model_family):
         q_embed = (q * cos) + (rotate_half(q) * sin)
         k_embed = (k * cos) + (rotate_half(k) * sin)
         return q_embed, k_embed
+    elif model_family == "gpt_neox":
+        gather_indices = position_ids[:, None, :, None]  # [bs, 1, seq_len, 1]
+        gather_indices = gather_indices.repeat(1, cos.shape[1], 1, cos.shape[3])
+        cos = torch.gather(cos.repeat(gather_indices.shape[0], 1, 1, 1), 2, gather_indices)
+        sin = torch.gather(sin.repeat(gather_indices.shape[0], 1, 1, 1), 2, gather_indices)
+        q_embed = (q * cos) + (rotate_half(q) * sin)
+        k_embed = (k * cos) + (rotate_half(k) * sin)
```

Comment:
These two implementations (gpt_neox and llama/baichuan) seems to be equivalent. see https://github.com/huggingface/transformers/commit/626c1b8af19f3d4cb2f2d2b116fdce45bed10ebb

Reply:
Correct, combined in a new PR, please take a look.